test/mechanics/solver/MumpsSolver.cpp

```diff
@@ -0,0 +1,42 @@
+#include "BoostUnitTest.h"
+#include "mechanics/solver/MumpsSolver.h"
+#include "base/Exception.h"
+#include <boost/test/data/test_case.hpp>
+#include <iostream>
+
+namespace bdata = boost::unit_test::data;
+
+using namespace NuTo;
+
+struct LinearSystem
+{
+    LinearSystem()
+    {
+        A.resize(3, 3);
+
+        // not very sparse, but hey...
+        A.insert(0, 0) = 1.0;
+        A.insert(0, 1) = -1.0;
+        A.insert(1, 0) = -1.0;
+        A.insert(1, 1) = 2.0;
+        A.insert(1, 2) = -1.0;
+        A.insert(2, 1) = -1.0;
+        A.insert(2, 2) = 2.0;
+
+        b << 3.0, -3.0, -2.0;
+        expected_x << 1.0, -2.0, -2.0;
+    }
+
+    Eigen::SparseMatrix<double> A;
+    Eigen::Vector3d b;
+    Eigen::Vector3d expected_x;
+};
+
+auto mumpsSolverNames = {"MumpsLU", "MumpsLDLT" };
+
+BOOST_DATA_TEST_CASE(mumpsSolvers, bdata::make(mumpsSolverNames), solver)
```

Comment:
1) First time I see the data test case, nice.

2) This test is very similar to `EigenSolver.cpp`. So maybe unite both? Or have a `LinearTestSystem.h`?

Reply:
1. Yeah, I liked that as well.
2. Can do.